git-remote-ipfs/src/Network/IPFS/Git/RemoteHelper.hs

```diff
@@ -64,6 +64,21 @@ renderProcessError = \case
     CidError  e       -> fmt ("Cid conversion error: " % fstr) e
     UnknownLocalRef r -> fmt ("Ref not found locally: " % ftxt) r
 
+newtype ConcurrentException =
+    ConcurrentException (RemoteHelperError ProcessError)
+
+instance Show ConcurrentException where
+    show (ConcurrentException e) =
+        sfmt ( "Network.IPFS.Git.RemoteHelper.ConcurrentException: "
+             % ftxt
+             % " "
+             % ftxt
+             )
+             (renderProcessError (errError e))
+             (renderSourceLoc    (errCallStack e))
+
+instance Exception ConcurrentException
```

Comment:
If you don't like the fact of implementing `Show` in a non-principled way, I guess you could try to implement `displayException` for `ConcurrentException` and see if that works out?

Reply:
I only recently realised the existence of `displayException`, and am confuse. `Show` is a superclass of `Exception`, so I still have to derive it for all error types - but that is something I try to avoid: one should always use the appropriate `render*` function.

Am I asking too much?

Reply:
Yeah, I am with you on this one. I guess the "best" thing you could do is to derive `Show` using the standing `deriving` mechanism but write your "pretty printed" version as part of `displayException`. I do agree that the superclass constraint is a nuisance.

Reply:
4b113d15e691d33bb79c52c698621b4f38e925bf destroyed my hopes